### PR TITLE
Migrate to Project.toml

### PR DESCRIPTION
**Pull request by davidanthoff**
Main reason is that I want to make sure the compat bounds on IteratorInterfaceExtensions and TableTraits are correct.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+IteratorInterfaceExtensions = "^0.1, ^1"
+OnlineStatsBase = "≥ 0.9.1"
+PooledArrays = "≥ 0.4.1"
+StructArrays = "≥ 0.3.2"
+WeakRefStrings = "≥ 0.5.7"
+julia = "≥ 1.0.0"
+TableTraits = "^0.3, ^0.4, ^1"
+
+[extras]
+Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
```
Comment (piever) [inline Project.toml]:
```suggestion
```

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+OnlineStats = "a15396b6-48d5-5d58-9928-6d29437db91e"
+PooledArrays = "2dfb63ee-cc39-5dd5-95bd-886bf059d720"
```
Comment (piever) [inline Project.toml]:
```suggestion
```

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+WeakRefStrings = "ea10d353-3f73-51f8-a26c-33c1cb351aa5"
```
Comment (piever) [inline Project.toml]:
```suggestion
```

File: Project.toml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name = "IndexedTables"
+uuid = "6deec6e2-d858-57c5-ab9b-e6ca5bd20e43"
+
+[deps]
+DataValues = "e7dc6d0d-1eca-5fa6-8ad6-5aecde8b7ea5"
+Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
```
Comment (piever) [inline Project.toml]:
```suggestion
```

File: Project.toml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+[extras]
+Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+OnlineStats = "a15396b6-48d5-5d58-9928-6d29437db91e"
+PooledArrays = "2dfb63ee-cc39-5dd5-95bd-886bf059d720"
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+WeakRefStrings = "ea10d353-3f73-51f8-a26c-33c1cb351aa5"
+
+[targets]
+test = ["Dates", "WeakRefStrings", "Test", "Random", "OnlineStats", "PooledArrays"]
```
Comment (piever) [inline Project.toml]:
```suggestion
test = ["Dates", "Test", "Random", "OnlineStats"]
```

**Comment by davidanthoff** [comment]
@piever I added your suggestions blindly :)

**Comment by piever** [comment]
:) Some requirements were duplicated, I don't know if that's what was causing the build to fail.

**Pull request merged**